src/core/EdFi.DataManagementService.Core/Middleware/RequestDataBodyLoggingMiddleware.cs

```diff
@@ -0,0 +1,100 @@
+// SPDX-License-Identifier: Apache-2.0
+// Licensed to the Ed-Fi Alliance under one or more agreements.
+// The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+// See the LICENSE and NOTICES files in the project root for more information.
+
+using System.Text.Json;
+using System.Text.RegularExpressions;
+using EdFi.DataManagementService.Core.Pipeline;
+using Microsoft.Extensions.Logging;
+
+namespace EdFi.DataManagementService.Core.Middleware;
+
+internal class RequestDataBodyLoggingMiddleware(ILogger _logger, bool _maskRequestBodyInLogs) : IPipelineStep
+{
+    private const string MessageBody =
+        "Incoming {Method} request to {Path} with body structure: {Body} - {TraceId}";
+
+    public async Task Execute(PipelineContext context, Func<Task> next)
+    {
+        if (_logger.IsEnabled(LogLevel.Debug) && !string.IsNullOrEmpty(context.FrontendRequest.Body))
+        {
+            _logger.LogDebug(
+                "Entering RequestDataBodyLoggingMiddleware - {TraceId}",
+                context.FrontendRequest.TraceId
+            );
+
+            string body = SanitizeInput(context.FrontendRequest.Body);
+
+            if (!_maskRequestBodyInLogs)
+            {
+                _logger.LogDebug(
+                    MessageBody,
+                    context.Method,
+                    context.FrontendRequest.Path,
+                    body,
+                    context.FrontendRequest.TraceId
+                );
+            }
+            else
+            {
+                _logger.LogDebug(
+                    MessageBody,
+                    context.Method,
+                    context.FrontendRequest.Path,
+                    MaskRequestBody(body, _logger),
+                    context.FrontendRequest.TraceId
+                );
+            }
+        }
+        await next();
+    }
+
+    private static string SanitizeInput(string input)
+    {
+        // Deletes new line, line feed, carriage return and tab characters
+        return Regex.Replace(
```

Comment:
I think this function is actually "minifying" the JSON, rather than "sanitizing" it.

Since you have a regex in there already, then it might be useful to create a single regex, and change to a [source-generated regular expression](https://learn.microsoft.com/en-us/dotnet/standard/base-types/regular-expression-source-generators?pivots=dotnet-8-0).

You might want to try out [this minification technique](https://stackoverflow.com/a/8913186/30384).